sdk/go/common/resource/plugin/testdata/analyzer-language/main.go

```diff
@@ -0,0 +1,155 @@
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type analyzer struct {
+	pulumirpc.UnimplementedAnalyzerServer
+}
+
+func (a *analyzer) Handshake(ctx context.Context, req *pulumirpc.AnalyzerHandshakeRequest) (*pulumirpc.AnalyzerHandshakeResponse, error) {
+	if req.StackConfiguration.Stack != "test-stack" {
+		return nil, fmt.Errorf("expected stack to be test-stack, got %s", req.StackConfiguration.Stack)
+	}
+	if req.StackConfiguration.Project != "test-project" {
+		return nil, fmt.Errorf("expected project to be test-project, got %s", req.StackConfiguration.Project)
+	}
+	if req.StackConfiguration.Organization != "test-org" {
+		return nil, fmt.Errorf("expected organization to be test-org, got %s", req.StackConfiguration.Organization)
+	}
+	if !req.StackConfiguration.DryRun {
+		return nil, fmt.Errorf("expected dry run to be true, got false")
+	}
+
+	actualConfig, err := plugin.UnmarshalProperties(req.StackConfiguration.Config, plugin.MarshalOptions{
+		KeepSecrets: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config: %w", err)
+	}
+
+	expectedConfig := resource.PropertyMap{
+		"test-project:bool":   resource.NewBoolProperty(true),
+		"test-project:float":  resource.NewNumberProperty(1.5),
+		"test-project:string": resource.NewStringProperty("hello"),
+		"test-project:obj": resource.NewObjectProperty(resource.PropertyMap{
+			"key": resource.NewStringProperty("value"),
+		}),
+	}
+
+	if !actualConfig.DeepEquals(expectedConfig) {
+		return nil, fmt.Errorf("expected config to be %v, got %v", expectedConfig, actualConfig)
+	}
+
+	return &pulumirpc.AnalyzerHandshakeResponse{}, nil
+}
+
+type language struct {
+	pulumirpc.UnimplementedLanguageRuntimeServer
+}
+
+func (l *language) GetPluginInfo(context.Context, *emptypb.Empty) (*pulumirpc.PluginInfo, error) {
+	return &pulumirpc.PluginInfo{
+		Version: "1.0.0",
+	}, nil
+}
+
+func (l *language) RunPlugin(req *pulumirpc.RunPluginRequest, srv pulumirpc.LanguageRuntime_RunPluginServer) error {
+	// This should be trying to run the analyzer plugin
+
+	if req.Kind != string(apitype.AnalyzerPlugin) {
+		return fmt.Errorf("expected kind to be ANALYZER, got %s", req.Kind)
+	}
+
+	// See if PULUMI_CONFIG= is in the environment
+	found := -1
+	for i, v := range req.Env {
+		if strings.HasPrefix(v, "PULUMI_CONFIG=") {
+			found = i
+			break
+		}
+	}
+	if found != -1 {
+		return fmt.Errorf("expected PULUMI_CONFIG not to be set, got %s", req.Env[found])
+	}
+	if req.Info.RootDirectory != req.Info.ProgramDirectory {
+		return fmt.Errorf("expected root directory to be the same as program directory, got %s and %s", req.Info.RootDirectory, req.Info.ProgramDirectory)
+	}
+	// Expect root directory to point to the policy pack
+	_, err := os.Stat(filepath.Join(req.Info.RootDirectory, "PulumiPolicy.yaml"))
+	if err != nil {
+		return fmt.Errorf("expected root directory to point to the policy pack, got %s", req.Info.RootDirectory)
```

Comment:
```suggestion
		return fmt.Errorf("missing PulumiPolicy.yaml in %s", req.Info.RootDirectory)
```

?

Reply:
Nah this is testdata, we know it should be pointing to the policypack and if the yaml isn't there its because the directory is wrong, not that the yaml is missing.